src/docgen/transpile/java.ts

```diff
@@ -0,0 +1,356 @@
+      typeReference: this.typeReference(property.type),
+      optional: property.optional,
+    };
+  }
+
+  public enum(enu: reflect.EnumType): transpile.TranspiledEnum {
+    return {
+      fqn: this.type(enu).fqn,
+      name: enu.name,
+    };
+  }
+
+  public enumMember(em: reflect.EnumMember): transpile.TranspiledEnumMember {
+    return {
+      fqn: `${this.enum(em.enumType).fqn}.${em.name}`,
+      name: em.name,
+    };
+  }
+
+  public unionOf(types: string[]): string {
+    return types.join(' OR ');
+  }
+
+  public listOf(type: string): string {
+    return `java.util.List<${type}>`;
+  }
+
+  public mapOf(type: string): string {
+    return `java.util.Map<java.lang.String, ${type}>`;
+  }
+
+  public any(): string {
+    return 'java.lang.Object';
+  }
+
+  public str(): string {
+    return 'java.lang.String';
+  }
+
+  public number(): string {
+    return 'java.lang.Number';
+  }
+
+  public boolean(): string {
+    return 'java.lang.Boolean';
+  }
+
+  public json(): string {
+    return 'com.fasterxml.jackson.databind.node.ObjectNode';
+  }
+
+  public date(): string {
+    return 'java.time.Instant';
+  }
+
+  public readme(readme: string): string {
+    return readme;
+  }
+
+  private formatImport(type: transpile.TranspiledType): string {
+    const namespace = type.namespace ? `.${type.namespace}` : '';
+    return `import ${type.module}${namespace}.${type.name};`;
+  };
+
+  private formatParameter(
+    transpiled: transpile.TranspiledParameter | transpile.TranspiledProperty,
+  ): string {
+    const tf = transpiled.typeReference.toString({
+      typeFormatter: (t) => t.name,
+    });
+    return `${tf} ${transpiled.name}`;
+  }
+
+  private formatInputs(inputs: string[]): string {
+    return inputs.join(', ');
+  };
+
+  private formatStructBuilder(type: transpile.TranspiledType, methods: string[]): string {
+    const builder = `${type.name}.builder()`;
+    return [
+      builder,
+      ...methods,
+      '    .build();',
+    ].join('\n');
+  };
+
+  private formatClassInitialization(
+    type: transpile.TranspiledType,
+    inputs: string[],
+  ): string {
+    return `new ${type.name}(${this.formatInputs(inputs)});`;
+  };
+
+  private formatClassBuilder(type: transpile.TranspiledType, parameters: reflect.Parameter[]): string {
+    const firstStruct: reflect.Parameter = parameters.find((param) => this.isStruct(param))!;
+    const struct: reflect.InterfaceType = firstStruct.parentType.system.findInterface(firstStruct.type.fqn!);
+    const positionalParams: reflect.Parameter[] = parameters.filter((param) => param.name !== firstStruct.name);
+    const createArgs: string = this.formatInputs(positionalParams.map((p) => this.formatParameter(this.parameter(p))));
+    const methods: string[] = struct.allProperties.map((p) =>
+      this.formatBuilderMethod(this.property(p)),
+    ).flat();
+    return [
+      `${type.name}.Builder.create(${createArgs})`,
+      ...methods,
+      '    .build();',
```

Comment:
The 4 spaces here seem magical - how do you guarantee its the same indentation as the builder methods? seems like this should be an argument that is passed to `formatBuilderMethod`

Reply:
OK - I refactored a bit to make the dependence more clear.